contract_manager/src/evm.ts

```diff
@@ -163,9 +271,36 @@ export class EVMContract extends Contract {
     };
   }
 
-  async executeGovernanceInstruction(privateKey: string, vaa: Buffer) {
-    const web3 = new Web3(this.chain.rpcUrl);
-    const { address } = web3.eth.accounts.wallet.add(privateKey);
+  async executeUpdatePriceFeed(senderPrivateKey: string, vaas: Buffer[]) {
+    const web3 = new Web3(this.chain.getRpcUrl());
+    const { address } = web3.eth.accounts.wallet.add(senderPrivateKey);
+    const pythContract = new web3.eth.Contract(EXTENDED_PYTH_ABI, this.address);
+    const priceFeedUpdateData = vaas.map((vaa) => "0x" + vaa.toString("hex"));
+    const updateFee = await pythContract.methods
+      .getUpdateFee(priceFeedUpdateData)
+      .call();
+    const transactionObject =
+      pythContract.methods.updatePriceFeeds(priceFeedUpdateData);
+    const gasEstiamte = await transactionObject.estimateGas({
+      from: address,
+      gas: 100000000,
+      value: updateFee,
+    });
+    let gasPrice = await web3.eth.getGasPrice();
+    if (!this.chain.isMainnet()) {
+      gasPrice = (BigInt(gasPrice) * 2n).toString();
+    }
+    return transactionObject.send({
+      from: address,
+      value: updateFee,
+      gas: gasEstiamte * 2,
```

Comment:
same comment on documenting why we do these.

Reply:
This was a dirty attempt to eliminate any need for manually specifying gas and gasPrice values per chain. For the `gasPrice` I was not sure how costly would it be to double the value on mainnet chains, so I only do it for testnet chains for now. If this is not an issue, I can do it everywhere and we don't need to worry a lot about manual entry of these values.
For the `gas` value, the `* 2` is just a very loose upper bound of how much gas is expected to be used.